apps/meteor/app/lib/server/functions/deleteMessage.ts

```diff
@@ -107,3 +107,22 @@ export async function deleteMessage(message: IMessage, user: IUser): Promise<voi
 		void bridges.getListenerBridge().messageEvent(AppEvents.IPostMessageDeleted, deletedMsg, user);
 	}
 }
+
+async function deleteThreadMessage(message: Required<Pick<IMessage, 'tmid'>>, user: IUser, room: IRoom | null): Promise<void> {
+	await Messages.decreaseReplyCountById(message.tmid, -1);
+
+	if (room) {
+		const { modifiedCount } = await Subscriptions.removeUnreadThreadsByRoomId(room._id, [message.tmid]);
+		if (modifiedCount > 0) {
+			void notifyOnSubscriptionChangedByRoomIdAndUserId(room._id, user._id);
+		}
+	}
+
+	// Check if this was the last reply in the thread
+	const thread = await Messages.findOneById(message.tmid, { projection: { tcount: 1 } });
```

Comment:
what do you think about modifying `decreaseReplyCountById` so as to use a `findOneAndUpdate` with the `returnDocument: 'after'` option?
This way we'd avoid this extra query here :eyes: